data/transactions/logic/kvcow.go

```diff
@@ -45,7 +79,20 @@ func (kvc *keyValueCow) read(key string) (value basics.TealValue, ok bool) {
 	return value, ok
 }
 
-func (kvc *keyValueCow) write(key string, value basics.TealValue) {
+func (kvc *keyValueCow) write(key string, value basics.TealValue) error {
+	// Enforce maximum key length
```

Comment:
It might be good to encapsulate these checks at some point (which duplicate checking in `teal.go`).